blackbird_python/blackbird/tests/test_program.py

```diff
@@ -422,6 +422,44 @@ def test_serialize_operation_multiple_arrays(self):
         )
         assert res == expected
 
+    def test_serialize_tdmprogram(self):
+        """Test serialization of a tdm program"""
+        bb = BlackbirdProgram(name="tdm", version=1.0)
+
+        # OrderedDict needed for Python 3.5 support
+        bb._type = OrderedDict(
+            [('name', 'tdm'), ('options', OrderedDict([('temporal_modes', 2), ('copies', 3)]))]
+        )
+        bb._var = OrderedDict(
+            [('BS', np.array([[1, 2]])), ('M', np.array([[3, 4]]))]
+        )
+        bb._operations.extend(
+            [
+                {'kwargs': {}, 'args': [0.7, 0], 'op': 'Sgate', 'modes': [1]},
+                {'kwargs': {}, 'args': ['BS', 0.0], 'op': 'BSgate', 'modes': [0, 1]},
+                {'kwargs': {'phi': 'M'}, 'args': [], 'op': 'MeasureHomodyne', 'modes': [0]}
+            ]
+        )
+
+        res = bb.serialize()
+        expected = dedent(
+            """\
+            name tdm
+            version 1.0
```

Comment:
Would we need a `target` field too here?

Reply:
Target isn't mandatory in a Blackbird script, so it's not needed (but wouldn't change anything).

Reply:
Could we have it in here? Just as a sanity check, as most TDM bb scripts would have it